### PR TITLE
Fix issue #1

### PR DESCRIPTION
**Pull request by TravisEz13**
Fix issue #1
add unit tests for issue #1
add a working sample to use for end to end tesing
Fix issue #3

### CODE REVIEW

File: Samples/SampleConfiguration.ps1 [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+     { 
+         Ensure = 'Present' 
+         ScopeID = '192.168.1.0' 
+         DnsDomain = 'contoso.com' 
+         DnsServerIPAddress = '192.168.1.22','192.168.1.1' 
+         AddressFamily = 'IPv4' 
+         Router = '192.168.1.1'
+         DependsOn = @('[WindowsFeature]DHCP') 
+     } 
+ }
```
Comment (vors) [inline Samples/SampleConfiguration.ps1]: add line to the end of file http://stackoverflow.com/questions/5813311/no-newline-at-end-of-file

File: DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.schema.mof [hunk 2]
```diff
@@ -1,10 +1,11 @@
...
-class MSFT_xDhcpServerOption : OMI_BaseResource
-{
-    [Key, Description("ScopeId for which options are set")] String ScopeID;
-    [Required, Description("IP address of DNS Servers")] String DnsServerIPAddress[];
-    [Write, Description("Domain name of DNS Server")] String DnsDomain;
-    [Write, Description("Address family type"), ValueMap{"IPv4"}, Values{"IPv4"}] String AddressFamily;
-    [Write, Description("Whether option should be set or removed"), ValueMap{"Present","Absent"}, Values{"Present","Absent"}] String Ensure;
-};
-
+[ClassVersion("1.0.0.0"), FriendlyName("xDhcpServerOption")]
```
Comment (vors) [inline DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.schema.mof]: What's the diff here? Is it in line endings?

File: TestSampleUsingAzure.ps1 [hunk 3]
```diff
@@ -0,0 +1,12 @@
+Write-Verbose -Message 'Publishing configuration ...' -Verbose
```
Comment (vors) [inline TestSampleUsingAzure.ps1]: Can you, please provide a comment how to run this end-to-end?

File: Tests/MSFT_xDhcpServerOption.tests.ps1 [hunk 4]
```diff
@@ -0,0 +1,222 @@
+<# 
+.summary
+    Test suite for MSFT_xDhcpServerOption.psm1
+    These tests require RSAT on client.
+    On 8.1 it's faund here: http://www.microsoft.com/en-us/download/confirmation.aspx?id=39296&6B49FDFB-8E5B-4B07-BC31-15695C5A2143=1
```
Comment (vors) [inline Tests/MSFT_xDhcpServerOption.tests.ps1]: typo `faund`

File: xDhcpServer.psd1 [hunk 5]
```diff
@@ -1,6 +1,6 @@
 @{
 # Version number of this module.
-ModuleVersion = '1.1'
+ModuleVersion = '1.1.1'
```
Comment (vors) [inline xDhcpServer.psd1]: Please, don't update module version https://github.com/PowerShell/DscResources/blob/master/CONTRIBUTING.md#creating-a-new-dsc-resource-in-an-existing-module

File: DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.schema.mof (hunk 2 above)
Comment (TravisEz13) [inline DSCResources/MSFT_xDhcpServerOption/MSFT_xDhcpServerOption.schema.mof]: line ending were wrong, corrected.

**Pull request merged**